### PR TITLE
Add `__unstable-large` size variant on `InputControl` `SelectControl` `UnitControl`

### PR DESCRIPTION
**Pull request by mirka**
Part of #36230

## Description

<details>
<summary>Previous iteration of this PR</summary>
<p>

Increases the size of the `default` InputControl to the new design (40px height with 16px horizontal paddings).

As part of this PR, we will audit all components and make the minimum necessary tweaks on those that depend on InputControl, so they do not look broken.

### Modified components (see commits for specifics)

- AnglePickerControl
- BoxControl
- ColorPicker
- CustomGradientPicker
- InputControl
- RangeControl
- SelectControl
- UnitControl

Aside from the components that were directly modified, the following components are also affected and checked for breakage:

- ✅ FocalPointPicker
- ✅ FontSizePicker with Slider
- ✅ GradientPicker
- ✅ NumberControl
- ✅ TreeSelect

⚠️ Components based on the [`input-control` mixin](https://github.com/WordPress/gutenberg/blob/enlarge-input-controls/packages/base-styles/_mixins.scss#L237) (e.g. TextControl, CustomSelectControl, ComboboxControl, etc.) will be dealt with in a separate PR.

## How has this been tested?

- Full audit of components in Storybook
- Both `default` and `small` variants look good
- Mobile viewports (where font-size increases to 16px)
- Looked through IRL component usage in the default block library for breakage. Follow-up tasks:
  - #35815

</p>
</details>

After some initial feedback and reconsideration, this PR now simply prepares an `__unstable-large` size variant on several basic components to allow opt-in upsizing in specific contexts. Default sizes are unchanged.

Breaking changes have been eliminated, and there are **no actual visual changes** being enabled in this PR. (Aside from a minor padding inconsistency fix in `UnitControl`.)

### Changes

| Component | Changes |
|---|---|
|   AnglePickerControl  |   🔧 tweaked internally (removed override styles and switched to unstable-large)  |
|   CustomGradientPicker  |   🔧 tweaked internally (removed override styles and switched to unstable-large)  |
|   InputControl 🧪  |   ✨ added unstable-large variant  |
|   ~RangeControl~  |   ~🔧 tweaked internally (accommodated Emotion class changes)~  |
|   SelectControl  |   ✨ added unstable-large variant  |
|   UnitControl 🧪  |   ✨ added unstable-large variant<br />  🐛 corrected inconsistent padding<br />   🔧 tweaked internally (simplified unit positioning to adapt to different heights)   |

## Next steps (separate PRs)

- #36193 (This is already 40px, we can remove the overrides and use unstable-large)
- More details in tracking issue: #36230

## How has this been tested?

- All size variants for `InputControl`, `SelectControl`, and `UnitControl` tested in Storybook.
- No visual regressions on all the changed components (see table above).

## Remaining tasks

- [x] Changelog entries
- [x] List out follow-up tasks

## Types of changes

New feature / Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Review by aaronrobertshaw — APPROVED** [review]
Thanks for all your attention to detail in making these updates @mirka 👍 

I've checked out all the, directly and indirectly, modified components as listed in the PR description. They all displayed consistently and the styling tweaks made all take effect. 

✅  AnglePickerControl
✅  BoxControl
✅  ColorPicker
✅  CustomGradientPicker
✅  InputControl
✅  RangeControl
✅  SelectControl
✅  UnitControl
✅  FocalPointPicker
✅  FontSizePicker with Slider
✅  GradientPicker
✅  NumberControl
✅  TreeSelect

I only had a couple of very minor questions.

1. ~The `CustomSelectControl` is in the follow-up list of controls based on the `input-control` mixin. I might have missed something but it looks like it could have been included in this batch of updates?~

2. Would it be better to have the `CHANGELOG` entry under `### Breaking Changes` rather than `### Experimental` given the type of changes stated for this PR was "Breaking change (styling)"?

Neither of those should block this PR landing at all though.

While testing the `ColorPicker` changes I saw some weird behaviour with it but that turned out to be unrelated to this PR and already has an [issue](https://github.com/WordPress/gutenberg/issues/36066).

**Comment by aaronrobertshaw** [comment]
I missed the follow-up, https://github.com/WordPress/gutenberg/pull/36162, regarding adjusting the size of the `CustomSelectControl` sorry. You can ignore that question in my review 🙂

**Comment by jasmussen** [comment]
Thank you for the PR. It's an important step. But in testing the PR I'm seeing a bit of a mix of things sizes now:
Screenshot 2021-11-03 at 09 58 59
Screenshot 2021-11-03 at 09 59 09

Screenshot 2021-11-03 at 09 59 17
Screenshot 2021-11-03 at 10 00 07

We know this is something we want, but due to how many things need to be updated, we need to find a way forward that works in small steps. Some ideas:

- Could we make the `small` version the default for now, and then flip the switch later?
- Do we need to make this a bigger PR and apply `small` manually to each input field we find that isn't yet ready for the new size?
- Can we reserve the new larger default size only for input fields used in a particular context, such as part of the new components?

**Comment by mirka** [comment]
Thank you so much for the reviews!

You both are right, I wasn't thinking clearly enough about how stable components rely on experimental components. Of the stable components this affects, `SelectControl` is one that I see being especially problematic. I'm going to rework this a bit using the approach in #36162 so we can roll out the changes in a more controlled way, and avoid breaking changes on stable components.

## Current overview

 🧪 experimental   ✔️ stable & non-breaking   ❗️stable & breaking

- AnglePickerControl ✔️
- BoxControl 🧪
- ColorPicker ✔️
- CustomGradientPicker ✔️
- InputControl 🧪
- RangeControl ❗️
- SelectControl ❗️
- UnitControl 🧪
- FocalPointPicker ❗️
- FontSizePicker ❗️
- GradientPicker ✔️
- NumberControl 🧪
- TreeSelect ❗️

> Would it be better to have the CHANGELOG entry under ### Breaking Changes rather than ### Experimental given the type of changes stated for this PR was "Breaking change (styling)"?

💯, I'll update this properly.

> Can we reserve the new larger default size only for input fields used in a particular context, such as part of the new components?

Will do.

**Comment by mirka** [comment]
I've scoped this down to be a preparatory PR — non-breaking and basically no visual changes. The PR description has been updated, and includes a list of next steps to selectively enable these large sizes.

@jasmussen Does this plan sound good to you?

**Comment by jasmussen** [comment]
I do like the sound of "no visual changes", especially if it allows you to move forward. I suggested [a bit of a crazy idea](https://github.com/WordPress/gutenberg/pull/36196#issuecomment-960498191) in your other PR, not sure if that's helpful here also.

**Comment by aaronrobertshaw** [comment]
> I do like the sound of "no visual changes", especially if it allows you to move forward. I suggested a bit of a crazy idea in your other PR, not sure if that's helpful here also.

I was of the understanding that all the separate PRs were to be merged together. This would effectively set all the controls to the `40px` height at the same time. Yesterday when testing, prior to the recent change in approach, I applied both PRs to see that the various inputs displayed consistently within the typography panel. Those controls that didn't were already flagged for updating in follow-ups.

File: packages/components/src/input-control/styles/input-control-styles.tsx
Comment (aaronrobertshaw) [inline packages/components/src/input-control/styles/input-control-styles.tsx]: I might be misinterpreting these changes here but weren't the default sizes to remain unchanged?

> After some initial feedback and reconsideration, this PR now simply prepares an __unstable-large size variant on several basic components to allow opt-in upsizing in specific contexts. Default sizes are unchanged.

**Review by aaronrobertshaw — COMMENTED** [review]
I've given this another test after the change in approach. 

Functionally everything is working however I did find some visual changes in terms of the height of some controls. In particular, the unit control. It appears that the tweaks to it have increased its height.

| Before | After |
|---|---|
| Screen Shot 2021-11-04 at 6 02 11 pm | Screen Shot 2021-11-04 at 5 52 10 pm |
| Screen Shot 2021-11-04 at 6 02 37 pm | Screen Shot 2021-11-04 at 5 53 01 pm |
| Screen Shot 2021-11-04 at 6 02 44 pm | Screen Shot 2021-11-04 at 5 53 07 pm |

File: packages/components/src/input-control/styles/input-control-styles.tsx
Comment (jasmussen) [reply packages/components/src/input-control/styles/input-control-styles.tsx]: I think we're moving towards having two sizes, 36px default, and 40px in some cases. At the moment, some of them are 30, some of them are 32, so it really is a bit all over the place, and so it feels okay to update some of these even if it does make for a few pixels of change here and there. For example most buttons are 36px, such as "Move to trash", which makes it really stand out next to a 30px input field.

In the other PR I wondered if we could use variables for these. We could set all values for both small and large to the most generic current value, such as 32. Then once we have a slew of these components updated, we can update both to 36 as a new baseline, and then one by one upgrade those that need it (notably in global styles and tools panels) to 40.

Comment (aaronrobertshaw) [reply packages/components/src/input-control/styles/input-control-styles.tsx]: > At the moment, some of them are 30, some of them are 32, so it really is a bit all over the place, and so it feels okay to update some of these even if it does make for a few pixels of change here and there.

Ok, I'm just confused. If it's alright that some controls are temporarily a few pixels out why couldn't they be a few pixels out and be set to `40px` straight away, thereby reducing the work and time to achieve the end result?

Comment (jasmussen) [reply packages/components/src/input-control/styles/input-control-styles.tsx]: Totally valid. In my opinion, 2-6px is a smallish change, and notably 36px would bring them in line with existing button sizes.

40px is a pretty big change, and a bold layout that only works when all counterparts are the same size.

But you're right though, if we can't touch controls like text-decoration toggle groups at the same time, we should definitely strive for "no visual changes" until such a time as we can do them all at the same time. But I'm unsure: could we use a variable to help us here?

Comment (aaronrobertshaw) [reply packages/components/src/input-control/styles/input-control-styles.tsx]: I'm not sure if a variable is feasible here given some components may apply the heights via hard coded styles. @mirka will be able to give a more informed opinion on that.

As for controls like the text-decoration toggle groups, I'd see that just being another PR in the "batch". When they are all in place and ready, they could all be merged. If we wanted to test them all together we could apply the patches for each PR on a single branch and confirm everything lines up nicely before hitting merge on anything.

Given all of Lena's hard work so far, I'd be inclined to support whatever approach she's most comfortable with.

Comment (mirka) [reply packages/components/src/input-control/styles/input-control-styles.tsx]: > I might be misinterpreting these changes here but weren't the default sizes to remain unchanged?

Argh great catch, you saved my life there! That was unintended, just a mangled revert — I confused it with the 36px default for button-like things. (I guess that's how confusing these size variants are, even after focusing on it for the past few days 😅)

Fixed now.

**Comment by aaronrobertshaw** [comment]
After the recent updates, everything looks good to me in both the block editor and storybook examples. I didn't spot any visual regressions compared to `trunk`.

<table>
  <tr>
    <td>Screen Shot 2021-11-08 at 11 07 23 am</td>
    <td>Screen Shot 2021-11-08 at 11 07 30 am</td>
    <td>Screen Shot 2021-11-08 at 11 06 48 am</td>
    <td>Screen Shot 2021-11-08 at 11 07 48 am</td>
  </tr>
</table>

**Pull request merged**
